cognee/tasks/memify/get_triplet_datapoints.py

```diff
@@ -0,0 +1,235 @@
+                logger.info(f"No more triplets found at offset {offset}. Processing complete.")
+                break
+
+            logger.debug(f"Retrieved {len(triplets_batch)} triplets in batch {batch_number}")
+
+            triplet_datapoints = []
+            skipped_count = 0
+
+            for idx, triplet_datapoint in enumerate(triplets_batch):
+                try:
+                    start_node = triplet_datapoint.get("start_node", {})
+                    end_node = triplet_datapoint.get("end_node", {})
+                    relationship = triplet_datapoint.get("relationship_properties", {})
+
+                    start_node_type = start_node.get("type")
+                    end_node_type = end_node.get("type")
+
+                    start_index_fields = datapoint_type_index_property.get(start_node_type, [])
+                    end_index_fields = datapoint_type_index_property.get(end_node_type, [])
+
+                    if not start_index_fields:
+                        logger.debug(
+                            f"No index_fields found for start_node type '{start_node_type}' in triplet {offset + idx}"
+                        )
+                    if not end_index_fields:
+                        logger.debug(
+                            f"No index_fields found for end_node type '{end_node_type}' in triplet {offset + idx}"
+                        )
+
+                    relationship_text = ""
+                    if relationship:
+                        edge_text = relationship.get("edge_text")
+                        if edge_text and isinstance(edge_text, str) and edge_text.strip():
+                            relationship_text = edge_text.strip()
+                        else:
+                            edge_type_index_fields = datapoint_type_index_property.get(
+                                "EdgeType", []
+                            )
+                            relationship_text = _extract_embeddable_text(
+                                relationship, edge_type_index_fields
+                            )
+
+                    start_node_text = _extract_embeddable_text(start_node, start_index_fields)
+                    end_node_text = _extract_embeddable_text(end_node, end_index_fields)
+
+                    start_node_id = start_node.get("id", "")
+                    end_node_id = end_node.get("id", "")
+
+                    if not start_node_id or not end_node_id:
+                        logger.warning(
+                            f"Skipping triplet at offset {offset + idx}: missing node IDs "
+                            f"(start: {start_node_id}, end: {end_node_id})"
+                        )
+                        skipped_count += 1
+                        continue
+
+                    embeddable_text = (
+                        f"{start_node_text}-›{relationship_text}-›{end_node_text}".strip()
+                    )
+
+                    if not embeddable_text:
```

Comment:
DIUC that `start_node_text` and `relationship_text` and `end_node_text` must be empty to enter this condition? In this case we check them before and skip the triplet, right? This condition seems unreachable